### ISSUE TITLE
Bring back 'renew' command

### ISSUE THREAD

**Issue by dougmoscrop** [Brightspace/bmx#179]
`bmx renew` did not survive the rewrite from python to golang I think it would be useful to support it.

Example:

`bmx renew --profile=test-foo`

Renew would:

- check if the given profile exists
- if it does, move on to checking if the credentials are still valid
-- if the credentials are valid, done
-- if expired, then refresh the credentials using the previously used account and role
- if it does not, act like `bmx write`

--profile would default to `$AWS_PROFILE` if unspecified.  I would argue that renew should be a profile-only command, and I actually completely discourage using default credentials _anywhere_

**Comment by chris-redekop** [comment]
From your description, I think a `--refresh` option on both `write` and `print` would make sense and be useful.  Did I get that right?

**Comment by dougmoscrop** [comment]
Yeah, although I think further to your point, `--refresh` should just be the default behaviour!

Or said, "As a developer, I want to type as few things as possible" 😆 

`bmx [write|print]` should:

- take $AWS_PROFILE if set
- override that with --profile when provided
- look to see what account/role was used for {resolved-profile}, if profile is being specified
- see if the credentials are expired, and refresh them if necessary
- write or print them depending on the command
